### PR TITLE
Keyword `ref`

### PR DESCRIPTION
**Pull request by AgniveshChaubey**
This PR addresses #49.

Changes requested in #43 have also been addressed.

### CODE REVIEW

File: content/2020-12/core/ref.markdown [hunk 1]
```diff
@@ -21,3 +21,196 @@ related:
   - vocabulary: core
     keyword: $defs
 ---
+
+The `$ref` keyword is used to statically reference a schema. This is useful for avoiding code duplication and promoting modularity when describing complex data structures.
+* The value of `$ref` is set to a URI reference, which may be either relative or absolute according to [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986).
+* A URI reference may include a [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) in a URI fragment (e.g., `#/foo/bar`), but it's important to note that these fragments apply to both relative and absolute URIs.
```
Comment (jviotti) [inline content/2020-12/core/ref.markdown:27]:
```suggestion
* A URI reference may include a [JSON Pointer](https://datatracker.ietf.org/doc/html/rfc6901) in a URI fragment (e.g., `#/foo/bar`)
```

This feels redundant, given you already said a URI reference may be absolute or relative

File: content/2020-12/core/ref.markdown [hunk 2]
```diff
@@ -21,3 +21,196 @@ related:
...
+
+{{<alert>}}
+ _**Note:** It's crucial to understand that an absolute URI does not necessarily denote a remote reference. An absolute URI can point to a local schema if the schema declares nested `$id` or if it points to itself. Conversely, a relative URI can point to a remote schema by leveraging base URI resolution._
```
Comment (jviotti) [inline content/2020-12/core/ref.markdown:30]:
```suggestion
 _**Note:** It's crucial to understand that an absolute URI does not necessarily denote a remote reference. An absolute URI can point to a local schema if the schema declares nested `$id`s or if it points to itself. Conversely, a relative URI can point to a remote schema by leveraging base URI resolution._
```

File: content/2020-12/core/ref.markdown [hunk 3]
```diff
@@ -21,3 +21,196 @@ related:
...
+{{<instance-pass `Instance including all the required properties is valid` >}}
+{
+  "productId": 123,
+  "name": "Widget"
+}
+{{</instance-pass>}}
+
+{{<instance-fail `Instance missing the required properties is invalid` >}}
+{
+  "name": "Gadget"
```
Comment (jviotti) [inline content/2020-12/core/ref.markdown:63]: I'm not sure I find this example great for this case. Maybe show off `name` being not a string?

File: content/2020-12/core/ref.markdown [hunk 4]
```diff
@@ -21,3 +21,196 @@ related:
...
+}
+{{</instance-fail>}}
+
+{{<schema `Schema with an absolute referece to the previous schema` >}}
```
Comment (jviotti) [inline content/2020-12/core/ref.markdown:67]:
```suggestion
{{<schema `Schema with an absolute reference to the previous schema` >}}
```

**Review by jviotti — APPROVED** [review]
I left a few minor comments, but pretty cool examples :+1:

File: content/2020-12/core/ref.markdown (hunk 3 above)
Comment (AgniveshChaubey) [reply content/2020-12/core/ref.markdown:63]: Sure!

**Pull request merged**